torch/csrc/jit/codegen/cuda/ir_iostream.cpp

```diff
@@ -19,7 +19,7 @@ namespace cuda {
 void checkInlineable(const Expr* expr) {
   for (auto input : expr->inputs()) {
     TORCH_CHECK(
-        input->isScalar() ||
+        input->isScalar() || input->isA<kir::TensorIndex>() ||
```

Comment:
In `torch.gather(input, dim, index)` operator, the selected index in `TensorIndex` of `input tensor` is another `TensorIndex`, which generated using `index tensor`. To support print this operator, we need the `TensorIndex` here.

Reply:
I guess this would allow basically everything in Kernel printed inline. What error did you have without this change?

Reply:
If we don't allow `TensorIndex` to print inline, some IR like `T1[ T2[idx] + ...]` cannot print.

Reply:
But, wouldn't it still print `T1`? Wouldn't it be just not inlined?

Reply:
It just gives me `Printing inline computations involving values other than scalars is not currently supported.` failure without this.

Reply:
Oh, I thought `checkInlineable` returns a bool indicating if it's possible to inline. Didn't realize it just fails if not.

What's the command to reproduce the error?

Reply:
Just remove this modification and run `PYTORCH_NVFUSER_DUMP=cuda_kernel,fusion_ir,lower_verbose ./build/bin/test_jit --gtest_filter='NVFuserTest.FusionTorchGatherIndexTvExtentIsOne_CUDA' `

Reply:
I think what @ftxj is doing here makes sense, in the `toString` method of `TensorIndex`, we have
```
std::string TensorIndex::toString(int indent_size) const {
  std::stringstream ss;
  ss << ir_utils::varName(this);
  switch (view()->getMemoryType()) {
    case MemoryType::Global:
      ss << "_g";
      break;
    case MemoryType::Shared:
      ss << "_s";
      break;
    case MemoryType::Local:
      ss << "_l";
      break;
    default:
      TORCH_INTERNAL_ASSERT(false, "Unknown tensor memory type.");
  }
  ss << "[";
  ss << index()->toInlineString(indent_size);
  ss << "]";
  ss << " view( " << ir_utils::varName(view()) << " )";
  return ss.str();
}
```
which prints the index inline. Without this change, we won't be able to print a `TensorIndex` object.